### PR TITLE
Derive solution template

### PR DESCRIPTION
**Pull request by novakov-alexey**

### CODE REVIEW

File: client/src/main/scala/client/SimpleWebSocketClient.scala [hunk 1]
```diff
@@ -8,37 +8,22 @@ import org.scalajs.dom.raw.MessageEvent
 import org.scalajs.dom.{CloseEvent, ErrorEvent, Event, WebSocket}
 
 import scala.concurrent.duration._
+import scala.language.postfixOps
 import scala.scalajs.js
+import scala.scalajs.js.Any
 
 final class SimpleWebSocketClient(url: String,
```
Comment (DmytroOrlov) [inline client/src/main/scala/client/SimpleWebSocketClient.scala]: let's revert whole class

File: client/src/main/scala/controller/AddTaskClient.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import monifu.reactive.OverflowStrategy.DropOld
+import org.scalajs.dom
+import shared.model._
+
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.scalajs.js.Dynamic.{literal => obj}
+import scala.scalajs.js.JSON
+
+class AddTaskClient(solutionCode: String) {
```
Comment (DmytroOrlov) [inline client/src/main/scala/controller/AddTaskClient.scala]: it is unused anymore, let's remove it?

File: server/app/controllers/AddTask.scala [hunk 3]
```diff
@@ -73,17 +82,31 @@ class AddTask @Inject()(executor: DynamicSuiteExecutor, dao: TaskDao, val messag
...
       }
     )
   }
+
+  def getSolutionTemplate: WebSocket = WebSocket.accept { req =>
+    def getTemplate(jsValue: JsValue) = Json.toJson(
+      SolutionTemplate(
+        Try(CodeParser.getSolutionTemplate((jsValue \ "solution").as[String]))
+          .toOption
+          .getOrElse("can't parse code")
```
Comment (DmytroOrlov) [inline server/app/controllers/AddTask.scala]: we need to return something different than `SolutionTemplate`

Comment (novakov-alexey) [inline server/app/controllers/AddTask.scala]: I think nothing else is needed to be sent. I just wrapped string into SolutionTemplate type to have some type-safety

File: server/app/controllers/JSONFormats.scala [hunk 4]
```diff
@@ -7,6 +7,7 @@ trait JSONFormats {
   private val lineFormat = Json.format[Line]
   private val testResultFormat = Json.format[TestResult]
   private val compilingFormat = Json.format[Compiling]
+  private val solutionTemplateFormat = Json.format[SolutionTemplate]
```
Comment (novakov-alexey) [inline server/app/controllers/JSONFormats.scala]: Presence of this class is a bit weird. I think Play should able to use Typeclasses to write and read JSON.

**Pull request merged**